src/components/molecules/NavBar/NavBar.tsx

```diff
@@ -276,62 +299,67 @@ export const NavBar: React.FC<NavBarPropsType> = ({ hasBackButton = true }) => {
                     placement="bottom-end"
                     overlay={
                       <Popover id="radio-popover">
-                        <Popover.Content>
-                          <RadioModal
-                            {...{
-                              volume,
-                              setVolume,
-                              title: currentVenue?.radioTitle,
-                            }}
-                            onEnableHandler={handleRadioEnable}
+                        <Popover.Content className="NavBar__radio--container">
+                          <RadioPopoverContent
+                            radioTitle={
+                              currentVenue?.radioTitle ?? "Playa Radio"
+                            }
+                            showNormalRadio={showNormalRadio}
+                            showSoundCloudRadio={showSoundCloudRadio}
+                            radioStation={radioStation}
                             isRadioPlaying={isRadioPlaying}
+                            handleRadioEnable={handleRadioEnable}
+                            volume={volume}
+                            setVolume={setVolume}
                           />
                         </Popover.Content>
                       </Popover>
                     }
                     rootClose={true}
                     defaultShow={showRadioOverlay}
                   >
-                    <button
-                      className={`profile-icon navbar-link-radio ${
-                        volume === 0 && "off"
-                      }`}
-                    />
+                    <button className="NavBar__menu--link">
+                      <img src={RadioIcon} alt="radio icon" />
+                    </button>
                   </OverlayTrigger>
                 )}
 
-                {showSoundCloudRadio && (
-                  <S.RadioTrigger>
-                    <button
-                      className={`profile-icon navbar-link-radio ${
-                        volume === 0 && "off"
-                      }`}
-                      onClick={toggleShowRadioPopover}
-                    />
-
-                    <S.RadioWrapper showRadioPopover={showRadioPopover}>
-                      <iframe
-                        title="venueRadio"
-                        id="sound-cloud-player"
-                        scrolling="no"
-                        allow="autoplay"
-                        src={`https://w.soundcloud.com/player/?url=${radioStation}&amp;start_track=0&amp;single_active=true&amp;show_artwork=false`}
-                      />
-                    </S.RadioWrapper>
-                  </S.RadioTrigger>
+                {showRadio && (
+                  <ButtonNG
+                    className={volumeControlClassname}
+                    onClick={toggleMute}
+                    iconOnly
+                  >
+                    <FontAwesomeIcon icon={volumeIcon} />
+                  </ButtonNG>
```

Comment:
`<ButtonNG>` has more props that can help streamline this code even further. Example:

```tsx
          <ButtonNG
            aria-label="Send message"
            className="Chatbox__emoji-button"
            onClick={toggleEmojiPicker}
            iconOnly={true}
            iconName={faSmile}
            iconSize="1x"
          />
```

Reply:
fixed 433dfe8